### PR TITLE
RELATE-6278: Adding blog post for single customer view

### PR DESCRIPTION
**Pull request by monitorjbl**

### CODE REVIEW

File: source/2022-07-15-single-customer-view/index.md [hunk 1]
```diff
@@ -0,0 +1,78 @@
+---
+title: Single Customer View
+tags: Genesys Cloud, Developer Engagement, Usage API
+date: 2022-07-15
+author: andrew.johnson
```
Comment (timsmithgenesys) [inline source/2022-07-15-single-customer-view/index.md:5]: @monitorjbl are you submitting this on Andrew's behalf?

**Review by timsmithgenesys — CHANGES_REQUESTED** [review]

File: source/2022-07-15-single-customer-view/index.md (hunk 1 above)
Comment (monitorjbl) [reply source/2022-07-15-single-customer-view/index.md:5]: Yeah, I am. Andy wrote the bulk of this, but didn't have time to blogify it so I'm doing that part for him.

File: source/2022-07-15-single-customer-view/index.md [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+image: genesys.png
```
[the remark this answers is not in the document]
Comment (monitorjbl) [reply source/2022-07-15-single-customer-view/index.md:6]: There is a PR up to add it in Bitbucket.

File: source/2022-07-15-single-customer-view/index.md [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+
+## Contact Types
+
+When the platform receives a new interaction on a supported channel, it resolves the identifiers for that interaction (for example: phone number, email address, social media identifier, etc.) to a contact.  If no matching contact is found, it creates one automatically.  It then sets the `participant.externalContactId` field on the [converesation object](https://developer.genesys.cloud/routing/conversations/conversations-apis#get-api-v2-conversations--conversationId-) to the ID of the found/created contact.
+
+These autogenerated contacts need to be distinguished from the contacts that your organization creates explicitly. To do this, we are introducing three types of contacts and a new `type` field on the Contact object.
+
+**Curated** contacts are the contacts you already know (and hopefully love). These are created via the API, the web application/desktop app, or via CSV/Salesforce imports. They never age out; you are responsible for their full lifecycle and eventual deletion just as you have always been.
```
[the remark this answers is not in the document]
Comment (monitorjbl) [reply source/2022-07-15-single-customer-view/index.md:30]: Good suggestion, will do.

File: source/2022-07-15-single-customer-view/index.md [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+
+Autogenerated contacts may be promoted to curated contacts to keep them in the curated set permanently. You may do this using the new [`POST /api/v2/externalcontacts/contacts/{contactId}/promotion`](/platform/preview-apis#post-api-v2-externalcontacts-contacts--contactId--promotion) API endpoint or the appropriate buttons in the Genesys Cloud web application. When you do so, the promoted contact will appear in all the places in the API that contacts currently do and you assume responsibility for their lifetime and compliance the same way you would any other contact you create.
+
+## Contact Merging
+
+Identity resolution can never be foolproof. A customer may have multiple devices that they use to contact your organization. Until they have been assigned to the contact that represents them, or if the interaction contains extra information that can be used to associate it to a known contact, the platform will assume that the new device is a new person. When this happens, some manual intervention will be required on the part of the agent.
```
Comment (andrewtjohnson) [inline source/2022-07-15-single-customer-view/index.md:39]: > ... or if the interaction contains extra information that can be used to associate it to a known contact, the platform will assume that the new device is a new person.

This seems incorrect.  If the interaction contains extra information that can be used to associate it to a known person, it *will not* create a new contact.  Guessing this is some sort of typo.

File: source/2022-07-15-single-customer-view/index.md [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+
+To do this, you can merge two contacts that represent the same individual using the [`POST /api/v2/externalcontacts/merge/contacts`](/platform/preview-apis#post-api-v2-externalcontacts-merge-contacts) endpoint. Merging two contacts creates a new “canonical” contact and joins all three into a “merge set.”  The canonical record holds the current state for the merge set. The other, non-canonical contacts in the merge set are called ancestor contacts and are made read-only. You can learn more about how to merge contacts and the types of contacts that may be merged in our [developer docs](/commdigital/externalcontacts/contact-merges#merging-contacts).
```
Comment (andrewtjohnson) [inline source/2022-07-15-single-customer-view/index.md:41]: To do "this" -- what does "this" refer to?  "some manual intervention" from the previous sentence?

File: source/2022-07-15-single-customer-view/index.md [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+
+## What does all this mean for you?
+
+Well, not much until your organization begins to use the merging features. Merging helps refine your organization's set of contacts over time and makes it easier to see all of the various interactions a customer has had. However, even without using the merge feature, your organization's agents can still get more visibility into your customer's past interactions by using the Customer Journey tab.
+
+Merging is gated behind a separate permission and endpoint, so it's not something that will happen unless your organization explicitly chooses to grant users permission to do it. Autogenerated contacts do not appear in search results or listing endpoints and should be largely invisible to most existing integrations. Before you start leveraging this functionality, make sure to carefully consider the impact as presented in this post.
```
Comment (andrewtjohnson) [inline source/2022-07-15-single-customer-view/index.md:71]: Change "grant users" to "grant agents"

**Review by andrewtjohnson — CHANGES_REQUESTED** [review]
Requested feedback in individual comments.

**Review by andrewtjohnson — CHANGES_REQUESTED** [review]
Left requested changes in comments.

File: source/2022-07-15-single-customer-view/index.md (hunk 6 above)
Comment (monitorjbl) [reply source/2022-07-15-single-customer-view/index.md:71]: Is that preferable? Not all users are agents.

File: source/2022-07-15-single-customer-view/index.md [hunk 7]
```diff
@@ -0,0 +1,75 @@
+---
+title: Single Customer View
+tags: Genesys Cloud, Developer Engagement, Usage API
```
[the remark this answers is not in the document]
Comment (monitorjbl) [reply source/2022-07-15-single-customer-view/index.md:3]: It was in the template, I think. Will remove it

File: source/2022-07-15-single-customer-view/index.md (hunk 5 above)
Comment (monitorjbl) [reply source/2022-07-15-single-customer-view/index.md]: Fixed

File: source/2022-07-15-single-customer-view/index.md (hunk 1 above)
Comment (timsmithgenesys) [reply source/2022-07-15-single-customer-view/index.md:5]: @andrewtjohnson no, posts can only have a single author.

File: source/2022-07-15-single-customer-view/index.md [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+
+For organizations that integrate with the Genesys Cloud platform, this can present some challenges. If your organization's integrations need to send data to a contact by its ID, that system would need to know the canonical contact ID in order to send the update to the correct record. While this can be done, it would be rather painful to force all users of this feature to have to update their integrations first.
+
+In order to ease the transition for such organizations, the API will transparently redirect some API operations made against ancestor contacts to the canonical contact of its merge set. Any of the following endpoints will exhibit this behavior:
+
+* [`GET /api/v2/externalcontacts/contacts/{contactId}`](https://developer.genesys.cloud/commdigital/externalcontacts/externalcontacts-apis#get-api-v2-externalcontacts-contacts--contactId-)
```
[the remark this answers is not in the document]
Comment (monitorjbl) [reply source/2022-07-15-single-customer-view/index.md]: Is there a way to build this site locally so I can see what it looks like in my document?

[the remark this answers is not in the document]
Comment (monitorjbl) [reply source/2022-07-15-single-customer-view/index.md]: Tim showed me a way to preview this. It looks like the choice is between this:

image

and this:

image

I prefer the latter, the former takes up a lot of space without much purpose. It's unlikely anyone reading this post will actually try to execute these endpoints directly from the page.

File: source/2022-07-15-single-customer-view/index.md [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+
+Several new concepts are being added to the platform API in order to support this new feature and we will explore them in this blog post together. Keep in mind that these new concepts should be non-disruptive to existing organizations; you will not be required to engage with most of these changes until your organization starts to use some of the new features described below.
+
+## Identifiers
+
+Until now, the way to look up contacts by identifiers like phone number, email address, etc. has been to use the [`GET /api/v2/externalcontacts/reversewhitepageslookup`](/commdigital/externalcontacts/externalcontacts-apis#get-api-v2-externalcontacts-reversewhitepageslookup) endpoint. This endpoint returns a list of contacts that contain an identifier with the provided lookup string.
```
Comment (andrewtjohnson) [inline source/2022-07-15-single-customer-view/index.md:15]: Do we need the backticks in the markdown links here?  Will that affect the rendering of links?

File: source/2022-07-15-single-customer-view/index.md (hunk 4 above)
Comment (andrewtjohnson) [reply source/2022-07-15-single-customer-view/index.md]: Let's talk through this on a call.  The revised version does not quite make sense to me, either.

Comment (andrewtjohnson) [reply source/2022-07-15-single-customer-view/index.md]: Oh, I think it's the command.  The middle statement in the original binds left, not right.  I think the comma in ", or if the interaction contains extra information" needs to be removed.  Ed can advise here.  I thought that clause went with the following statement, which would make it incorrect.

Maybe there's a way to reword the whole thing to make it clearer and put all the exception stuff last after an "unless".

<normal thing that happens> unless <exception case 1> <exception case 2>.

File: source/2022-07-15-single-customer-view/index.md (hunk 9 above)
Comment (timsmithgenesys) [reply source/2022-07-15-single-customer-view/index.md:15]: Need? No. You're making a style choice here. Links and inline code are rendered the same in github vs dev center. Examples:

With backticks: [`GET /api/v2/externalcontacts/reversewhitepageslookup`](/commdigital/externalcontacts/externalcontacts-apis#get-api-v2-externalcontacts-reversewhitepageslookup)

Without: [GET /api/v2/externalcontacts/reversewhitepageslookup](/commdigital/externalcontacts/externalcontacts-apis#get-api-v2-externalcontacts-reversewhitepageslookup)

File: source/2022-07-15-single-customer-view/index.md [hunk 10]
```diff
@@ -0,0 +1,75 @@
...
+
+When contacts are created using the [`POST /api/v2/externalcontacts/contacts`](/commdigital/externalcontacts/externalcontacts-apis#post-api-v2-externalcontacts-contacts) endpoint, the platform attempts to claim Identifiers that correspond to each of the named identifier fields (such as cellPhone, workEmail, etc.).  If any identifier is already claimed by another contact, it remains claimed by that contact. As contacts are edited via the [`PUT`](/commdigital/externalcontacts/externalcontacts-apis#put-api-v2-externalcontacts-contacts--contactId-) and [`DELETE`](/commdigital/externalcontacts/externalcontacts-apis#delete-api-v2-externalcontacts-contacts--contactId-) endpoints, the identifiers corresponding to the contact are claimed/released appropriately. 
+
+You can read more about identifiers and their associated operations in our [developer documentation](/commdigital/externalcontacts/contact-merges#identifiers).
```
Comment (andrewtjohnson) [inline source/2022-07-15-single-customer-view/index.md:21]: testing what "Start a review" means

File: source/2022-07-15-single-customer-view/index.md [hunk 11]
```diff
@@ -0,0 +1,75 @@
...
+
+## Contact Types
+
+When the platform receives a new interaction on a supported channel, it resolves the identifiers for that interaction (for example: phone number, email address, social media identifier, etc.) to a contact.  If no matching contact is found, it creates one automatically.  It then sets the `participant.externalContactId` field on the [converesation object](/routing/conversations/conversations-apis#get-api-v2-conversations--conversationId-) to the ID of the found/created contact.
```
Comment (andrewtjohnson) [inline source/2022-07-15-single-customer-view/index.md:25]: here is a review comment

**Review by andrewtjohnson — CHANGES_REQUESTED** [review]
here is my overall review comment

**Review by andrewtjohnson — APPROVED** [review]

File: source/2022-07-15-single-customer-view/index.md [hunk 12]
```diff
@@ -0,0 +1,75 @@
...
+
+Autogenerated contacts may be promoted to curated contacts to keep them in the curated set permanently. You may do this using the new [`POST /api/v2/externalcontacts/contacts/{contactId}/promotion`](/platform/preview-apis#post-api-v2-externalcontacts-contacts--contactId--promotion) API endpoint or the appropriate buttons in the Genesys Cloud web application. When you do so, the promoted contact will appear in all the places in the API that contacts currently do and you assume responsibility for their lifetime and compliance the same way you would any other contact you create.
+
+## Contact Merging
+
+Identity resolution can never be foolproof. A customer may have multiple devices that they use to contact your organization. Unless those devices have been assigned to the contact that represents the person (or if the interaction contains extra information that can be used to associate it to an existing contact), the platform will assume that the new device is a new person and create a new, potentially duplicate contact. When this happens, some manual intervention will be required on the part of the agent.
```
Comment (andrewtjohnson) [inline source/2022-07-15-single-customer-view/index.md:39]: unless (A or B)
vs
(unless A) or B

Comment (andrewtjohnson) [reply source/2022-07-15-single-customer-view/index.md:39]: unless either (A or B), then

File: source/2022-07-15-single-customer-view/index.md (hunk 9 above)
Comment (monitorjbl) [reply source/2022-07-15-single-customer-view/index.md:15]: Yeah, I personally like the inline code style for this so I went with it for all the endpoint links. It makes it more obvious that the entire phrase is together when linebreaks split it up. Plus, a monospaced font screams "this is for developers," so the eyes of more technically-minded folk will be drawn to these bits of of the document.

**Review by carnellj-genesys — APPROVED** [review]

**Pull request merged**